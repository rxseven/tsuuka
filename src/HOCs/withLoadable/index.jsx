import LoadableVisibility from 'react-loadable-visibility/react-loadable';

import Loader from 'components/common/composite/Loader';

function withLoadable(loader) {
  return LoadableVisibility({
    delay: 200,
    loader,
    loading: Loader,
    timeout: 10000
  });
}

export default withLoadable;
