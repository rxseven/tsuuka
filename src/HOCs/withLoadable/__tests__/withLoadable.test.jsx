import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import withLoadable from '../index';

describe('withLoadable HOC', () => {
  // Arrange
  const component = <withLoadable />;

  it('should render without crashing', () => {
    // Act
    const wrapper = shallow(component);

    // Assertions
    expect(wrapper).toBeDefined();
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
