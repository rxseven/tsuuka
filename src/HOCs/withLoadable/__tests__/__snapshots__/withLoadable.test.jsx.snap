// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLoadable HOC Snapshot tests should render correctly 1`] = `<withLoadable />`;
